boards/BMS/Src/App/states/App_ChargeState.c

```diff
@@ -25,18 +34,42 @@ static void ChargeStateRunOnTick100Hz(struct StateMachine *const state_machine)
     struct BmsWorld *world = App_SharedStateMachine_GetWorld(state_machine);
     struct BmsCanTxInterface *can_tx  = App_BmsWorld_GetCanTx(world);
     struct Charger *          charger = App_BmsWorld_GetCharger(world);
+    struct Clock *            clock   = App_BmsWorld_GetClock(world);
 
     if (!App_Charger_IsConnected(charger))
     {
         App_CanTx_SetPeriodicSignal_CHARGER_DISCONNECTED_IN_CHARGE_STATE(
             can_tx, true);
         App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
     }
+
+    if (App_Charger_HasFaulted(charger))
+    {
+        App_CanTx_SetPeriodicSignal_CHARGER_FAULT_DETECTED(can_tx, true);
+        App_SharedStateMachine_SetNextState(state_machine, App_GetFaultState());
+    }
+
+    // Track the time since accumulator charging started
+    const uint32_t elapsed_time_s =
+        App_SharedClock_GetCurrentTimeInSeconds(clock) -
+        App_SharedClock_GetPreviousTimeInSeconds(clock);
+
+    // Go to the fault state when charging has completed to indicate that
+    // the charger should be disconnected and the BMS-Core should be
+    // power-cycled
+    if (elapsed_time_s >= TOTAL_CHARGE_TIME_S)
```

Comment:
is this how we are defining when we have reached 100% soc? I think this may not be the best way to determine that we are fully charged up

Reply:
I think doing a check to see if the MAX voltage has reached 4.2V or something might be a better approach instead of using a time limit

Reply:
agreed